packages/xo-server-audit/src/index.js

```diff
@@ -0,0 +1,222 @@
+import appConf from 'app-conf'
+import asyncIteratorToStream from 'async-iterator-to-stream'
+import createLogger from '@xen-orchestra/log'
+import path from 'path'
+import { alteredAuditRecord, missingAuditRecord } from 'xo-common/api-errors'
+import { fromCallback } from 'promise-toolbox'
+import { pipeline } from 'readable-stream'
+import {
+  AlteredRecordError,
+  AuditCore,
+  MissingRecordError,
+  NULL_ID,
+  Storage,
+} from '@xen-orchestra/audit-core'
+
+const log = createLogger('xo:xo-server-audit')
+
+const LAST_ID = 'lastId'
+class Db extends Storage {
+  constructor(db) {
+    super()
+    this._db = db
+  }
+
+  async put(record) {
+    await this._db.put(record.id, record)
+  }
+
+  async del(id) {
+    await this._db.del(id)
+  }
+
+  get(id) {
+    return this._db.get(id).catch(error => {
+      if (!error.notFound) {
+        throw error
+      }
+    })
+  }
+
+  async setLastId(id) {
+    await this._db.put(LAST_ID, id)
+  }
+
+  getLastId() {
+    return this.get(LAST_ID)
+  }
+}
+
+const NAMESPACE = 'audit'
+class AuditXoPlugin {
+  constructor({ xo }) {
+    this._cleaners = []
+    this._xo = xo
+
+    this._auditCore = undefined
+    this._blockedList = undefined
+    this._storage = undefined
+  }
+
+  async load() {
+    const cleaners = this._cleaners
+
+    try {
+      const storage = (this._storage = new Db(
+        await this._xo.getStore(NAMESPACE)
+      ))
+      this._auditCore = new AuditCore(storage)
+      this._blockedList = (
+        await appConf.load('xo-server-audit', {
```

Comment:
`xo-server-audit` is not a true application but a plugin, it does not really makes sense to use `app-conf` for it.

That's why we are not doing this for other plugins, we should fix this in the next step, please note it.

Reply:
[Done](https://github.com/vatesfr/xen-orchestra/issues/4798) :+1: